Assets/_Project/Scripts/Map/LocationProviderMapUpdater.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using Mapbox.Unity.Location;
+using Mapbox.Unity.Map;
+using Mapbox.Utils;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath.Map
+{
+    public class LocationProviderMapUpdater : MonoBehaviour
+    {
+        [SerializeField] private AbstractMap _map;
+        
+        private LocationFactory _locationFactory;
+        
+        private bool _isMapInitialized;
+
+        private bool _initialized;
+
+        private Coroutine _deviceLocationPollCoroutine;
+
+        private bool isPolling = false;
+        
+        public LocationFactory LocationFactory
+        {
+            get
+            {
+                return _locationFactory;
+            }
+            private set
+            {
+                _locationFactory = value;
+            }
+        }
+
+        public void Initialize(LocationFactoryMode mode)
+        {
+            
+            _initialized = true;
```

Comment:
TBH not sure if it matters but it looks better when a flag is set when everything is really done (move it to the bottom).